.github/workflows/blackhole-post-commit.yaml

```diff
@@ -14,16 +14,22 @@ permissions:
   contents: read
   pages: write
   id-token: write
+  packages: write
 
 jobs:
   static-checks:
     uses: ./.github/workflows/all-static-checks.yaml
     secrets: inherit
+  build-docker-image:
```

Comment:
why not just push this into `build-artifact`?

Reply:
Currently build-artifact will always build in docker, but it can be toggled to build a docker image as part of build-artifact or to assume the image already exists.

It's written in the current case where we enable more tests (ie profiler) all require the same docker image, so i just left it as build the docker image external to build-artifact so we avoid having to make this structure in the future.